### DISCUSSION TITLE
api call for uptime returns plaintext not json

### DISCUSSION THREAD

**Discussion by PoleCatRoad** [nicolargo/glances#2025] [Q&A] [answered]
Calling glances api by curl yields this plaintext result:

http://0.0.0.0:61208/api/3/uptime: "1 day, 15:45:28"

Whereas other calls return json as expected:

http://0.0.0.0:61208/api/3/core: {"phys": 6, "log": 12}
http://0.0.0.0:61208/api/3/core/phys: {"phys": 6}

Peculiarly, 'glances --issue' gives json result:

uptime        [OK]    0.00004s {'seconds': 140725}

I notice that 2 other api fields that have only a single value ('psutilversion' and 'now') return plaintext both by curl and --issue.  As far as I can tell only 'uptime' returns different data types in curl vs --issue.  Caveat:  I have not tried ALL of the possible api calls so I don't know all where this occurs.

What is the reason for this to happen?  I saw nothing in the api documentation that would explain this. The '--issue' option isn't mentioned in the docs that I could see.  I only knew to use it from an earlier question here.  There may be other undocumented options/switches.

Q: Is there a way for user to receive from the api a json response for uptime?  

No responses in official google group since Sep 2021 so I post this here.

Regards,
fitz

**Comment by PoleCatRoad** [comment, marked as answer]
It seems glances must run a task, maybe at startup, that gets the public ip.  This means it would not change what is reported if/when my isp changes my ip until glances restarts.

cat /proc/uptime show uptime in seconds, which is probably where --issue gets it.  The simple command 'uptime' returns the formatted information so this is reported via api.  Why they are handled differently I can't guess.  So just those minor confusions in the documentation.  

This concludes my several hour trip into proc and psutil.

**Reply by skynetforcats** [reply]
I was just looking for this too. Why would I want uptime in a formatted string?
But anyway did you find a way to get a timestamp or seconds from the API? I hate to reverse format that string myself.

**Reply by PoleCatRoad** [reply]
Not from the glances api but my workaround is to ssh into each system and run 'uptime -s' which returns the formatted datetime since the system booted.  Reverse format that string into what I want.   Sorry it's been so long since I chacked back here.
